src/main/java/org/springframework/data/elasticsearch/client/reactive/ReactiveElasticsearchClient.java

```diff
@@ -411,6 +412,41 @@
 	 */
 	Flux<SearchHit> search(HttpHeaders headers, SearchRequest searchRequest);
 
+	/**
+	 * Execute the given {@link SearchRequest} with aggregations against the {@literal search} API.
+	 *
+	 * @param consumer never {@literal null}.
+	 * @see <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html">Search API on
+	 *      elastic.co</a>
+	 * @return the {@link Flux} emitting {@link Aggregation} one by one.
+	 */
+	default Flux<Aggregation> aggregate(Consumer<SearchRequest> consumer) {
+
```

Comment:
```suggestion
    Assert.notNull(consumer, "consumer must not be null"):
```

Parameters that are not annotated as `@Nullable` must be checked.